typeset/converter.js

```diff
@@ -106,129 +109,70 @@ const injectMathJax = async (log, inputPath, cssPath, outputPath, mathJaxPath) =
-      })
-      console.log(`Attempting to inject MathJax from "${mathJaxPath}"`)
-      mjax.src = mathJaxPath
-      document.body.appendChild(mjax)
-    })
-  }, mathJaxPath)
-
-  if (!didMathJaxLoad) {
-    log.fatal('MathJax did not load')
-    return STATUS_CODE.ERROR
-  }
-  await sleep(1000) // wait for MathJax to load
-
-  log.info(`Polling to see when MathJax is done typesetting...`)
-  let pageContentAfterSerialize = ''
-  while (true) {
-    const {isFailed, isDone} = await page.evaluate(/* istanbul ignore next */() => {
-      if (!window.MathJax) {
-        console.error('MathJax was not loaded')
-        return {isFailed: true}
-      }
-      let msg = document.getElementById('MathJax_Message')
-      if (msg && msg.innerText !== '') {
-        console.info(`Progress: "${document.getElementById('MathJax_Message').innerText}"`)
-      }
-      return {
-        isDone: window.__TYPESET_CONFIG.isDone,
-        isFailed: window.__TYPESET_CONFIG.isFailed
-      }
-    })
-    if (isFailed) {
-      log.fatal('Failed for some reason. Check logs')
-      await browser.close()
-      return STATUS_CODE.ERROR
-    } else if (isDone) {
-      log.info('Serializing document...')
-      pageContentAfterSerialize = await page.evaluate(/* istanbul ignore next */() => {
-        // Remove any elements we added
-        window.__TYPESET_CONFIG.elementsToRemove.forEach(el => el.remove())
-
-        let s = new window.XMLSerializer()
-        let str = s.serializeToString(document)
-        if (window.MathJax && window.MathJax.Message) {
-          console.log('All messages from MathJax:', JSON.stringify(window.MathJax.Message.Log()))
-        } else {
-          console.error('Could not find window.MathJax.Message')
-        }
-        return str
-      })
-      break
+    /* Array.prototype.forEach.call(mathMLElements, (el, i) => {
+      mathMLElementsMap[i] = el.outerHTML
+      el.outerHTML = `<span id="mjnode-${i}" class="mjnode-replace"></span>`
+      console.log(`mathMLElements.length: ${mathMLElements.length}`)
+      console.log(`Switched ${i} element`)
+    }) */
+    /* for(let i = 0; i < mathMLElements.length; i++){
+      mathMLElementsMap[i] = mathMLElements[i].outerHTML
+      mathMLElements[i].outerHTML = `<span id="mjnode-${i}" class="mjnode-replace"></span>`
+      console.log(`Switched ${i} element`)
+    } */
+    console.log(`Found after ${mathMLElements.length} <m:math> elements`)
+
+    console.log(`All elements replaced with numbered span.`)
+
+    console.log('Serializing content...')
+    let s = new window.XMLSerializer()
+    let serializedContent = s.serializeToString(document)
+
+    let res = {
+      serializedContent: serializedContent,
+      mathMLElementsMap: mathMLElementsMap
     }
+    return res
+  })
 
-    // Wait a second before polling again
-    await sleep(1000)
-  }
+  log.info('Saving file without MathML elements ...')
+  await writeFile(output, res.serializedContent)
```

Comment:
The file is saved without Math elements and then reopened once `mathjax-node` is done converting. Why not just keep it open? 

Also, if mathjax-node fails for some reason it seems like we might accidentally assume the conversion succeeded since the "file without MathML elements" is saved to the output file location. Would it make sense to create a temporary file instead?

Two downsides of creating a temporary file are:is that it is harder to debug

Reply:
I'm not sure why but when I didn't save file then in then next `page.evalute()` I was'nt able to find `<span class="mjnode-replace">..`. 

I will double check on this.